tests/system/Database/Builder/SelectTest.php

```diff
@@ -67,6 +67,21 @@ public function testSelectAcceptsArray(): void
         $this->assertSame($expected, str_replace("\n", ' ', $builder->getCompiledSelect()));
     }
 
+    public function testSelectAcceptsArrayWithRawSql(): void
```

Comment:
Try to reverse the order and you won't get a good query, the "employee_id" column will be missing:

```php
        $builder->select([
            new RawSql("IF(salary > 5000, 'High', 'Low') AS salary_level"),
            'employee_id',
        ]);
```

```sql
-'SELECT IF(salary > 5000, 'High', 'Low') AS salary_level, "employee_id" FROM "employees"'
+'SELECT IF(salary > 5000, 'High', 'Low') AS salary_level FROM "employees"'
```

Reply:
Fixed.

Reply:
I think it's not work again, try:

```php

$builder->select([
      new RawSql("IF(salary > 5000, 'High', 'Low') AS salary_level"),
      new RawSql("IF(test > 1, 'YES', 'NO') AS yes_no"),
      'employee_id',
]);
```
```sql
-'SELECT IF(salary > 5000, 'High', 'Low') AS salary_level, IF(test > 1, 'YES', 'NO') AS yes_no, "employee_id" FROM "employees"'
+'SELECT IF(salary > 5000, 'High', 'Low') AS salary_level, IF(salary > 5000, 'High', 'Low') AS salary_level, "employee_id" FROM "employees"'
```

Reply:
Fixed.